src/main/java/com/uber/cadence/internal/worker/PollTaskExecutor.java

```diff
@@ -81,6 +81,11 @@ public void process(T task) {
         });
   }
 
+  @Override
+  public boolean hasCapacity() {
+    return taskExecutor.getQueue().remainingCapacity() > 0;
```

Comment:
I think this need to be `taskExecutor.getActiveCount() < taskExecutor.getPoolSize()`